### PR TITLE
 [fix](nereids) fix wrong result precision for add/sub

### PR DESCRIPTION
**Pull request by jacktengg**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by jacktengg** [comment]
run buildall

**Comment by jacktengg** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/Add.java [hunk 1]
```diff
@@ -49,7 +49,7 @@ public Expression withChildren(List<Expression> children) {
 
     @Override
     public DecimalV3Type getDataTypeForDecimalV3(DecimalV3Type t1, DecimalV3Type t2) {
-        return (DecimalV3Type) DecimalV3Type.widerDecimalV3Type(t1, t2, false);
+        return (DecimalV3Type) DecimalV3Type.widerDecimalV3TypeForAddSub(t1, t2, false);
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/Add.java:52]: do not change widerDecimalV3TypeForAddSub directly, use widerDecimalV3TypeForAddSub's result and plus 1 on its precision to create a new DecimalV3Type object

**Comment by jacktengg** [comment]
run buildall

**Comment by jacktengg** [comment]
run buildall

**Comment by jacktengg** [comment]
run feut

**Comment by jacktengg** [comment]
run feut

**Review by morrySnow — APPROVED** [review]

**Review by starocean999 — APPROVED** [review]

**Pull request merged**